File: src/test/testData/inspections/AnonymousInnerClassLeak.java
import android.app.Activity;
import android.os.Handler;

class AnonymousInnerClassLeak extends Activity {
    void runTask() {
        // Bad: Anonymous inner class captures outer activity
        new Handler().postDelayed(<warning descr="LeakLens: Anonymous inner class holds an implicit reference to AnonymousInnerClassLeak. If this object outlives the Activity, it will prevent GC.">new Runnable() {
            public void run() {
                // no-op
            }
        }</warning>, 1000);
    }

    // Good: Static inner class does not capture outer
    static class MySafeRunnable implements Runnable {
        public void run() {}
    }

    void runSafeTask() {
        new Handler().post(new MySafeRunnable());
    }
}

File: src/test/testData/inspections/AnonymousInnerClassLeak.kt
import android.app.Activity
import android.os.Handler

class AnonymousInnerClassLeakKt : Activity() {
    fun runTask() {
        // Bad: Object literal captures outer activity
        Handler().postDelayed(< warning descr =
            "LeakLens: Anonymous inner class holds an implicit reference to AnonymousInnerClassLeakKt. If this object outlives the Activity, it will prevent GC." > object :
                Runnable {
                override fun run() {
                    // no-op
                }
            } < / warning >, 1000)
    }

    // Good: Nested class (static by default)
    class MySafeRunnable : Runnable {
        override fun run() {}
    }

    fun runSafeTask() {
        Handler().post(MySafeRunnable())
    }
}
